scipy/stats/tests/test_morestats.py

```diff
@@ -835,6 +839,56 @@ def test_1d_input(self):
         assert_raises(ValueError, stats.levene, g1, x)
 
 
+class TestFEqualityTest:
+
+    def test_data1(self):
+        args = [f1, f2]
+        T, pval = f_equality_test(*args, alternative='two-sided')
+        assert_almost_equal(T, 1.207788039, 7)
+        assert_almost_equal(pval, 0.7105517535985708, 7)
+
+    def test_data2(self):
+        args = [f1, f2]
+        T, pval = f_equality_test(*args, alternative='greater')
+        assert_almost_equal(T, 1.207788039, 7)
+        assert_almost_equal(pval, 0.6447241232007151, 7)
+
+    def test_data3(self):
+        args = [f1, f2]
+        T, pval = f_equality_test(*args, alternative='less')
+        assert_almost_equal(T, 1.207788039, 7)
+        assert_almost_equal(pval, 0.3552758767992854, 7)
+
+    def test_data4(self):
+        args = [f1, f2]
+        T, pval = f_equality_test(*args)
+        assert_almost_equal(T, 1.207788039, 7)
+        assert_almost_equal(pval, 0.7105517535985708, 7)
+
+    def test_bad_arg(self):
+        args = [f1, f2]
+        # Too few args raises ValueError.
+        assert_raises(ValueError, f_equality_test, [1])
+        # Alternative not valid raises ValueError.
+        assert_raises(ValueError, f_equality_test, *args, 'twosided')
+        assert_raises(ValueError, f_equality_test, *args, 'lower')
+
+    def test_result_attributes(self):
+        args = [f1, f2]
+        res = f_equality_test(*args)
+        attributes = ('statistic', 'pvalue')
+        check_named_results(res, attributes)
+
+    def test_empty_arg(self):
+        args = (f1, [])
+        assert_equal((np.nan, np.nan), f_equality_test(*args))
+
+    # temporary fix for issue #9252: only accept 1d input
+    def test_1d_input(self):
+        x = np.array([[1, 2], [3, 4]])
+        assert_raises(ValueError, f_equality_test, f1, x)
```

Comment:
I don't think this is needed. You applied the `@_axis_nan_policy` decorator, so this will handle N-D inputs with an `axis` argument, `nan_policy`, etc. if it is configured correctly. We can work on that later.
```suggestion
```